controller-server/src/main/java/com/yahoo/vespa/hosted/controller/persistence/CuratorDb.java

```diff
@@ -67,7 +73,23 @@
 
     @Inject
     public CuratorDb(Curator curator) {
+        this(curator, curator.zooKeeperEnsembleConnectionSpec());
+    }
+
+    CuratorDb(Curator curator, String zooKeeperEnsembleConnectionSpec) {
         this.curator = curator;
+        this.zooKeeperEnsembleConnectionSpec = zooKeeperEnsembleConnectionSpec;
+    }
+
+    /** Returns all hosts configured to be part of this ZooKeeper cluster */
+    public List<HostName> cluster() {
+        return Arrays.stream(zooKeeperEnsembleConnectionSpec.split(","))
+                     .filter(hostPort -> !hostPort.isEmpty())
+                     .map(hostPort -> hostPort.contains(":")
+                             ? hostPort.substring(0, hostPort.indexOf(":"))
+                             : hostPort)
```

Comment:
I think `.map(host -> host.replaceFirst(":.*", ""))` is clearer, but feel free to disagree.

Reply:
Fixed.